.ci/pitest.groovy

```diff
@@ -0,0 +1,322 @@
+                break
+        }
+    }
+    if (lineContent == null) {
+        String mutationFileName = mutationClassPackage + sourceFile
+        String startingPath = ".${SEPARATOR}src${SEPARATOR}main${SEPARATOR}java${SEPARATOR}"
+        String javaExtension = ".java"
+        mutationFileName = startingPath + mutationFileName
+                .substring(0, mutationFileName.length() - javaExtension.length())
+                .replaceAll("\\.", SEPARATOR) + javaExtension
+
+        File file = new File(mutationFileName)
+        lineContent = XmlUtil.escapeXml(file.readLines().get(lineNumber - 1).trim())
+    }
+    if (lineNumber == 0) {
+        lineNumber = -1
+    }
+    boolean unstable = false
+    String unstableMutation = mutationNode.attribute("unstable")
+    if (unstableMutation != null) {
+        unstable = Boolean.parseBoolean(unstableMutation)
+    }
+    return new Mutation(sourceFile, mutatedClass, mutatedMethod, mutator, description,
+            lineContent, lineNumber, unstable)
+}
+
+private static int compareSurvivedAndSuppressedMutations(Set<Mutation> survivedMutations,
+                                                         Set<Mutation> suppressedMutations) {
+
+    int exitCode = 0
+    if (survivedMutations != suppressedMutations) {
+        Set<Mutation> survivedAndUnsuppressedMutations =
+                setDifference(survivedMutations, suppressedMutations)
+        if (!survivedAndUnsuppressedMutations.isEmpty()) {
+            exitCode = 1
+            println "Surviving mutation/s found:"
+            survivedAndUnsuppressedMutations.each { println it }
+        }
+        Set<Mutation> killedAndSuppressedMutations =
+                setDifference(suppressedMutations, survivedMutations)
+        if (!killedAndSuppressedMutations.isEmpty()) {
+            Mutation stableUnnecessarySuppression = killedAndSuppressedMutations.find {
+                return !it.unstable()
+            }
+            if (stableUnnecessarySuppression != null) {
+                exitCode = 1
+                println "\nUnnecessary suppressed mutations/s found:"
+                killedAndSuppressedMutations.each {
+                    if (!it.unstable()) {
+                        println it
+                    }
+                }
+            }
+        }
+    }
+    return exitCode
+}
+
+private static Set<Mutation> setDifference(final Set<Mutation> setOne,
+                                           final Set<Mutation> setTwo) {
+    Set<Mutation> result = new HashSet<Mutation>(setOne)
+    result.removeIf(mutation -> setTwo.contains(mutation))
+    return result
+}
+
+record Mutation(String sourceFile, String mutatedClass,
+                String mutatedMethod, String mutator,
+                String description, String lineContent, int lineNumber,
+                boolean unstable)
+        implements Comparable<Mutation> {
+
+    private static final int LINE_NUMBER_NOT_PRESENT_VALUE = -1
+
+    boolean equals(o) {
```

Comment:
Please use `other` or `that` for parameter name. Do we not need `@Override` here and on hashcode below?

Reply:
Done, added `@Override` too.